File: GGAndroid_2.0/GGControls/GGViewModel/src/main/java/io/ganguo/viewmodel/ui/dialog/FullScreenVModelDialog.java
package io.ganguo.viewmodel.ui.dialog;

import android.content.Context;
import android.databinding.ViewDataBinding;

import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelDialog;

/**
 * <p>
 * 全屏弹窗 - dialog
 * </p>
 * Created by leo on 2018/9/30.
 */
public abstract class FullScreenVModelDialog<T extends ViewDataBinding, B extends BaseViewModel> extends ViewModelDialog<T, B> {
    public FullScreenVModelDialog(Context context) {
        super(context);
    }

    @Override
    protected boolean isFullScreen() {
        return true;
    }
}

File: GGAndroid_3.0/core/frame/mvvm-v1/src/main/java/io/ganguo/viewmodel/pack/ui/dialog/FullScreenVModelDialog.kt
package io.ganguo.viewmodel.pack.ui.dialog

import android.content.Context
import android.view.ViewGroup
import androidx.databinding.ViewDataBinding
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.view.ViewModelDialog

/**
 *
 *
 * 全屏弹窗 - dialog
 *
 * Created by leo on 2018/9/30.
 */
abstract class FullScreenVModelDialog<T : ViewDataBinding, B :  BaseViewModel<*>>(context: Context, theme: Int) : ViewModelDialog<T, B>(context, theme) {
    override val dialogWidth: Int
        protected get() = ViewGroup.LayoutParams.MATCH_PARENT

    override val dialogHeight: Int
        protected get() = ViewGroup.LayoutParams.MATCH_PARENT
}